arquillian-environment-reporter/arquillian-environment-reporter-impl/src/main/java/org/arquillian/environment/reporter/impl/DockerDetector.java

```diff
@@ -0,0 +1,26 @@
+package org.arquillian.environment.reporter.impl;
+
+import java.io.File;
+
+public class DockerDetector {
+
+    static final String DOCKER_SOCK = "docker.sock";
+    static final String DOCKERINIT  = ".dockerinit";
+    static final String DOCKERENV   = ".dockerenv";
+
+    static final String ROOT_DOCKER_FILE   = "/";
+    static final String ROOT_DOCKER_SOCKET = "/var/run/";
+
+    static final File DOCKER_ENV_PATH = new File(ROOT_DOCKER_FILE, DOCKERENV);
+    static final File DOCKER_INIT_PATH = new File(ROOT_DOCKER_FILE, DOCKERINIT);
+    static final File DOCKER_SOCKET_FILE = new File(ROOT_DOCKER_SOCKET, DOCKER_SOCK);
+
+    private DockerDetector() {
+        super();
```

Comment:
Sure, you just ensure that java spec is not broken ;)

> If the subclass constructor does not specify which superclass constructor to invoke then the compiler will automatically call the accessible no-args constructor in the superclass.

The thing is - there is no superclass here besides the ancestor of all the classes :)